ckanext/datastore/logic/action.py

```diff
@@ -420,7 +422,10 @@ def datastore_search(context, data_dict):
     :param language: language of the full text query
                      (optional, default: english)
     :type language: string
-    :param limit: maximum number of rows to return (optional, default: 100)
+    :param limit: maximum number of rows to return
+        (optional, default: ``100``, unless set in the site's configuration
+        ``ckan.datastore.search.rows_default``, upper limit: ``32000`` unless
+        set in site's configuration ``ckan.datastore.search.rows_max``)
```

Comment:
Should have raised this in my last comment, but If we're adding the hard upper limit why do we need a default limit? AFAICT this just breaks external users scripts for no real benefit.

We've got a pretty big overhead serving `datastore_search` calls and if a user sees that they're only getting 100 records they might start paginating themselves 100 records at a time, creating *way* more load on the server. Is there a reason you don't want to just rely on the new rows_max setting?

Reply:
The default of 100 has been there since 2012 (https://github.com/ckan/ckan/commit/3dec0e8#diff-5816a6ed066765bf38ec73a848d96e3aR150), so I don't think we're breaking anything, are we? I think if someone is going to seek out the 'offset' parameter to achieve paging, then they'll likely see the 'limit' too and increase that value to make their life easier. So arguably on balance, removing the default might increase load overall. However I can also see that maybe it's an unnecessary complication having both default and max.

Reply:
you're right! nevermind then